File: cpp/se/ArDomainPolicyUndeleteCommand.cpp
#include "se/ArDomainPolicyUndeleteCommand.hpp"
#include "se/ArExtension.hpp"
#include "se/ArDomainObjectType.hpp"
#include "se/CommandType.hpp"
#include "xml/XMLHelper.hpp"
#include "xml/XStr.hpp"

namespace {
    class ArPolicyUndeleteCommandType : public CommandType
    {
        public:
            ArPolicyUndeleteCommandType() : CommandType(getCommandName()) { }
            std::string getCommandName() const { return "policyUndelete"; }
            std::string toString() const { return "policyUndelete"; }
    };

    Extension& arExtension() {
        static Extension* arExt = new ArExtension();
        return *arExt;
    }

    const ArPolicyUndeleteCommandType polUndeleteCmdType;
    const ArDomainObjectType ardomType;
}; // anonymous namespace

ArDomainPolicyUndeleteCommand::ArDomainPolicyUndeleteCommand(
        const std::string &name) : ProtocolExtensionCommand(
            &polUndeleteCmdType, &ardomType, name, arExtension())
{
}

File: src/main/java/com/ausregistry/jtoolkit2/se/ArDomainPolicyUndeleteCommand.java
package com.ausregistry.jtoolkit2.se;

/**
 * Mapping of EPP urn:ar:params:xml:ns:arext-1.0 policy domainUndelete command
 * specified by the AusRegistry EPP extensions document.  This should only be
 * used to request reversing the deleting of domains which have been deleted due to policy
 * violation using the policy delete operation.
 * Use this class to generate an AusRegistry-compliant XML document, given
 * simple input parameters.  The toXML method in Command serialises this object
 * to XML.
 */
public class ArDomainPolicyUndeleteCommand extends ProtocolExtensionCommand {
    private static final long serialVersionUID = 420815971854196895L;

    private static final CommandType POLICY_UNDELETE_COMMAND_TYPE = new ArPolicyUndeleteCommandType();

    public ArDomainPolicyUndeleteCommand(String name) {
        super(POLICY_UNDELETE_COMMAND_TYPE, ExtendedObjectType.AR_DOMAIN, name);
    }

    @Override
    protected Extension getExtension() {
        return ExtensionImpl.AR;
    }
}

